examples_RPi/transfer_interrupt.cpp

```diff
@@ -0,0 +1,192 @@
+ the slower form of high-speed transfer using blocking-writes.
+ */
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <RF24/RF24.h>
+#include <unistd.h>
+
+using namespace std;
+//
+// Hardware configuration
+//
+
+/****************** Raspberry Pi ***********************/
+
+// Radio CE Pin, CSN Pin, SPI Speed
+// See http://www.airspayce.com/mikem/bcm2835/group__constants.html#ga63c029bd6500167152db4e57736d0939 and the related enumerations for pin information.
+
+// Setup for GPIO 22 CE and CE0 CSN with SPI Speed @ 4Mhz
+//RF24 radio(RPI_V2_GPIO_P1_22, BCM2835_SPI_CS0, BCM2835_SPI_SPEED_4MHZ);
+
+// NEW: Setup for RPi B+
+//RF24 radio(RPI_BPLUS_GPIO_J8_15,RPI_BPLUS_GPIO_J8_24, BCM2835_SPI_SPEED_8MHZ);
+
+// Setup for GPIO 15 CE and CE0 CSN with SPI Speed @ 8Mhz
+RF24 radio(RPI_V2_GPIO_P1_15, RPI_V2_GPIO_P1_24, BCM2835_SPI_SPEED_8MHZ);
+
+/*** RPi Alternate ***/
+//Note: Specify SPI BUS 0 or 1 instead of CS pin number.
+// See http://tmrh20.github.io/RF24/RPi.html for more information on usage
+
+//RPi Alternate, with MRAA
+//RF24 radio(15,0);
+
+//RPi Alternate, with SPIDEV - Note: Edit RF24/arch/BBB/spi.cpp and  set 'this->device = "/dev/spidev0.0";;' or as listed in /dev
+//RF24 radio(22,0);
+
+
+/****************** Linux (BBB,x86,etc) ***********************/
+
+// See http://tmrh20.github.io/RF24/pages.html for more information on usage
+// See http://iotdk.intel.com/docs/master/mraa/ for more information on MRAA
+// See https://www.kernel.org/doc/Documentation/spi/spidev for more information on SPIDEV
+
+// Setup for ARM(Linux) devices like BBB using spidev (default is "/dev/spidev1.0" )
+//RF24 radio(115,0);
+
+//BBB Alternate, with mraa
+// CE pin = (Header P9, Pin 13) = 59 = 13 + 46 
+//Note: Specify SPI BUS 0 or 1 instead of CS pin number. 
+//RF24 radio(59,0);
+
+/**************************************************************/
+
+// Radio pipe addresses for the 2 nodes to communicate.
+const uint64_t addresses[2] = { 0xABCDABCD71LL, 0x544d52687CLL };
+
+
+uint8_t data[32];
+unsigned long startTime, stopTime, counter, rxTimer=0;
+
+void intHandler()
+{
+	//Read as long data is available
+	//Single interrupts may be lost if a lot of data comes in.
+	while(radio.available())
+	{
+      radio.read(&data,32);
+      counter++;
+    }
+}
+
+int main(int argc, char** argv){
+
+  bool role_ping_out = 1, role_pong_back = 0;
+  bool role = 0;
+
+  // Print preamble:
+
+  cout << "RF24/examples/Transfer/\n";
+
+  radio.begin();                           // Setup and configure rf radio
+  radio.setChannel(1);
+  radio.setPALevel(RF24_PA_MAX);
+  radio.setDataRate(RF24_1MBPS);
+  radio.setAutoAck(1);                     // Ensure autoACK is enabled
+  radio.setRetries(2,15);                  // Optionally, increase the delay between retries & # of retries
+  radio.setCRCLength(RF24_CRC_8);          // Use 8-bit CRC for performance
+  radio.printDetails();
```

Comment:
I just realized also that adding ```radio.maskIRQ(1,1,0);``` in examples that only use the interrupt for reading will prevent the radio from triggering and interrupt on TX_FAIL and TX_OK. Going to test this more with RF24Gateway and the mutex locking etc.